src/impl.cpp

```diff
@@ -0,0 +1,17 @@
+#include "impl.h"
+
+int BinaryInvent (int x, int p, int n)
+{
+    int ii{}, temp{};
```

Comment:
please never use "temp/tmp" in names, on stack everything is "temp"
also, please follow the codestyle and don't write variables in one line